packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -140,7 +137,11 @@ function NavigationMenuSelector( {
 		},
 	};
 
-	if ( ! hasNavigationMenus && ! hasClassicMenus ) {
+	if (
+		hasResolvedNavigationMenus &&
```

Comment:
Added explicit conditional that we don't render this until the menus have resolved.